### PR TITLE
Use new string functions instead of strpos().

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Console/BaseCommand.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ abstract class BaseCommand implements CommandInterface
      */
     public function setName(string $name)
     {
-        if (strpos($name, ' ') < 1) {
+        if (!strpos($name, ' ')) {
```
Comment (othercorey) [inline src/Console/BaseCommand.php:48]: That's an odd check. Might be cleaner to trim it first.

File: src/Utility/Text.php [hunk 2]
```diff
@@ -801,7 +801,7 @@ protected static function _substr(string $text, int $start, ?int $length, array
             $len = self::_strlen($part, $options);
             if ($offset !== 0 || $totalLength + $len > $length) {
                 if (
-                    strpos($part, '&') === 0
+                    mb_strpos($part, '&') === 0
```
Comment (othercorey) [inline src/Utility/Text.php]: Should the preg_ functions be multi-byte or unicode?

**Comment by othercorey** [comment]
That is a lot. They all look good. The comments I left are probably separate PRs.

File: src/Console/BaseCommand.php (hunk 1 above)
Comment (markstory) [inline src/Console/BaseCommand.php:48]:
```suggestion
        if (!str_contains($name, ' ')) {
```

Isn't this a 'not contains' check?

Comment (ADmad) [reply src/Console/BaseCommand.php:48]: Can't just trim it off, there's a specific test which throws an exception for leading space.

File: src/Utility/Text.php (hunk 2 above)
Comment (markstory) [reply src/Utility/Text.php]: The pattern on 785 is looking for HTML entities and hexadecimal entities. Those patterns shouldn't require unicode support.

Comment (ADmad) [reply src/Utility/Text.php]: IDK, I just changed this one as other functions in the method where `mb_*`.

File: src/Console/BaseCommand.php (hunk 1 above)
Comment (othercorey) [reply src/Console/BaseCommand.php:48]: Apparently, there is a special check for starting with a space for some reason.

Comment (ADmad) [reply src/Console/BaseCommand.php:48]: @markstory `!str_contains()` will pass for string with leading space ` foo` which we don't want to allow. There's also a test for that specific case.

Comment (markstory) [reply src/Console/BaseCommand.php:48]: Oh fun.

**Pull request merged**